### ISSUE TITLE
`wlroots-hyprland-hidpi` fails to build on NixOS

### ISSUE THREAD

**Issue by colin-heffernan** [hyprwm/Hyprland#1694]
Please consult the issue guidelines at
https://github.com/vaxerski/Hyprland/blob/main/docs/ISSUE_GUIDELINES.md
BEFORE submitting.

Specifically, Hunk #7 fails when patching `xwayland/xwm.c`.
```zsh
building the system configuration...
error: builder for '/nix/store/b64jiwbbksqm6xayjxhm2dqljsflkhv8-wlroots-hyprland-hidpi-2023-03-02_5ae17de.drv' failed with exit code 1;
       last 10 log lines:
       > patching file xwayland/xwm.c
       > Hunk #1 succeeded at 19 with fuzz 1 (offset -1 lines).
       > Hunk #2 succeeded at 98 (offset 1 line).
       > Hunk #3 succeeded at 977 (offset 20 lines).
       > Hunk #4 succeeded at 1009 (offset 20 lines).
       > Hunk #5 succeeded at 1027 (offset 20 lines).
       > Hunk #6 succeeded at 1144 (offset 21 lines).
       > Hunk #7 FAILED at 1732.
       > Hunk #8 succeeded at 2149 (offset 79 lines).
       > 1 out of 8 hunks FAILED -- saving rejects to file xwayland/xwm.c.rej
       For full logs, run 'nix log /nix/store/b64jiwbbksqm6xayjxhm2dqljsflkhv8-wlroots-hyprland-hidpi-2023-03-02_5ae17de.drv'.
error: 1 dependencies of derivation '/nix/store/5pblz93q6m7l19g4glz5gjzs19f02f2z-hyprland-0.22.0+date=2023-03-03_c121706.drv' failed to build
error (ignored): error: cannot unlink '/tmp/nix-build-eww-8ff4142064afae76397ba11c9a7bafc10d2dc4d1.drv-0': Directory not empty
error: 1 dependencies of derivation '/nix/store/24vn815qcm1kcgzcp1k3392wmj70yd4s-desktops.drv' failed to build
error: 1 dependencies of derivation '/nix/store/kdi32l1h5pvkc3a5v0r5i58k4brj6wcf-hyprland-share-picker-0.pre+date=2023-01-07_d479c84.drv' failed to build
error: 1 dependencies of derivation '/nix/store/cxazmjacklacpbhpcbxq4vyrvfpf0cjy-system-path.drv' failed to build
error: 1 dependencies of derivation '/nix/store/zindfdj2cg77d3aqsxcv0br0wzachs73-nixos-system-heffos-obsidian-23.05.20230301.a08d697.drv' failed to build
```

To reproduce, try to install Hyprland via the Nix Flake on NixOS Unstable.
Expected: Hyprland and its dependencies build correctly.
Noted: `wlroots-hyprland-hidpi` fails to build.

I can submit any other info as necessary.

**Comment by vaxerski** [comment]
@fufexan for nix

**Comment by fufexan** [comment]
The fix is almost ready.

**Comment by colin-heffernan** [comment]
Thanks for the quick response. I can test the fix whenever it's ready!

**Comment by Eken-beep** [comment]
I get the same thing but with `nix log /nix/store/jqj4hic9ps2qh90zmh369gbx0mwkq6mk-wlroots-hyprland-hidpi-nvidia-2023-03-02_5ae17de.drv`

**Comment by vaxerski** [comment]
@fufexan shouldnt this now be closado?

**Comment by fufexan** [comment]
Yes. Probably didn't autoclose because I didn't write `fixes #1694` directly.
